File: CTrove/angularapp/src/app/models/contributor/contributor.ts
import { ADD, ORGANIZATION, SEARCH, STATUS } from 'src/Utilities/common/app-strings';
import { Base, BaseFilter, Deserializable } from '../dto/common';
import { ContributorStudy } from './contributor-study';
import { Country } from './country';
import { Organization } from './organization';

export class Contributor extends Base implements Deserializable {
  objectId!: string;
  countryId!: string;
  country!: Country;
  email!: string;
  city!: string;
  prefix!: number;
  firstname!: string;
  lastname!: string;
  grade!: number;
  primaryJobTitle!: string;
  secondaryJobTitle!: string;
  phone!: string;
  mobile!: string;
  publicData!: boolean;
  organizationId!: string;
  organization!: Organization;
  active!: boolean;
  consent!: boolean;
  dateOfConsent!: Date;
  ContributorStudy!: ContributorStudy[];

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}

export class ContributorFilter extends BaseFilter implements Deserializable {
  countrysId!: string[];
  organizationsId!: string[];

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}

// Fields
export const FIELDS = {
  search: SEARCH,
  prefix: 'Prefix',
  firstName: 'First Name',
  middleName: 'Middle Name',
  lastName: 'Last Name',
  grade: 'Grade',
  email: 'Email',
  mobileNo: 'Mobile Number',
  landLine: 'Landline',
  organization: ORGANIZATION,
  publicData: 'Public Data',
  consent: 'Consent',
  dateOfConsent: 'Date of Consent',
  studies: 'Studies',
  country: 'Country',
  city: 'City',
  primaryJob: 'Primary Job',
  secondaryJob: 'Secondary Job',
  status: STATUS,
  add: ADD
}

File: CTrove.Core/Entity/Contributor.cs
using CTrove.Core.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CTrove.Core.Entity
{
    public class Contributor
    {
        public Guid Id { get; set; }
        public Guid ObjectId { get; set; }
        public Guid CountryId { get; set; }
        public virtual Country Country { get; set; }
        public string Email { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public PrefixType Prefix { get; set; }
        //public string? Prefix { get; set; } = string.Empty;
        public string Firstname { get; set; } = string.Empty;
        public string Lastname { get; set; } = string.Empty;

        public GradeType Grade { get; set; }
       // public string? Grade { get; set; } = string.Empty;
        public string? PrimaryJobTitle { get; set; } = string.Empty;
        public string? SecondaryJobTitle { get; set; } = string.Empty;
        public string? Phone { get; set; } = string.Empty;
        public string? Mobile { get; set; } = string.Empty;
        public bool PublicData { get; set; }
        public Guid OrganizationId { get; set; }
        public virtual Organization Organization { get; set; }
        public bool Active { get; set; }
        public bool Consent { get; set; }
        public DateTime? DateOfConsent { get; set; }
        public bool Status { get; set; }
        public virtual ICollection<ContributorStudy> ContributorStudy { get; set; }
    }
}
